aptos-move/framework/aptos-stdlib/sources/type_info.move

```diff
@@ -23,6 +25,18 @@ module aptos_std::type_info {
 
     public native fun type_name<T>(): string::String;
 
+    /// Return the BCS size, in bytes, of value at `val_ref`.
+    ///
+    /// See the [BCS spec](https://github.com/diem/bcs)
+    ///
+    /// See `test_size_of_val()` for an analysis of common types and
+    /// nesting patterns, as well as `test_size_of_val_vectors()` for an
+    /// analysis of vector size dynamism.
+    public fun size_of_val<T>(val_ref: &T): u64 {
```

Comment:
I wonder if we should rename this to "serialized_size_of_val" for more clarity.

Reply:
non-serialize version would be called "size_of"?

Reply:
@vgao1996 @lightmark per @davidiw 's [above comment](https://github.com/aptos-labs/aptos-core/pull/4540#issuecomment-1295173518) this PR has been renamed `size_of_val` after previously having been named `size_of`, with the latter more appropriately referring to a native function that accepts only type arguments.

I yield to @davidiw and @wrwg for guidance on renaming to `serialized_size_of_val`, which I am glad to do if there is consensus.